File: 백준/Bronze/1193. 분수찾기/분수찾기.py
def cal(i):
    n=i*(i+1)/2
    return n

X=int(input())
i=1
if(1<=X and X<=10000000):
    while(1):
        if(cal(i-1)<=X and X<=cal(i)):
            break
        i+=1
    if(i%2==0):
        cnt=cal(i)-X
        a=int(i-cnt)
        b=int(cnt+1)
    elif(i%2!=0):
        cnt=cal(i)-X
        a=int(cnt+1)
        b=int(i-cnt)
    print(a,b,sep='/')

File: 백준/Bronze/1193. 분수찾기/분수찾기.c
#include<stdio.h>
int cal(int i);

int main() {
	int X,i=1,cnt=0,a=0,b=0;
	scanf("%d", &X);
	if (1 <= X && X <= 10000000) {
		while (1) {
			if (cal(i - 1) <= X && X <= cal(i)) {
				break;
			}
			i++;
		}
		if (i % 2 == 0) {
			cnt = cal(i) - X;
			a = i - cnt;
			b = cnt + 1;
		}
		else if (i % 2 != 0) {
			cnt = cal(i) - X;
			a = cnt + 1;
			b = i - cnt;
		}
	}
	printf("%d/%d", a, b);
	return 0;
}

int cal(int i) {
	int n;
	n = i * (i + 1) / 2;
	return n;
}